packages/components/Menu/Menu.tsx

```diff
@@ -22,51 +23,56 @@ export const Menu: React.FC<MenuProps> = ({
   component,
   width = DEFAULT_WIDTH,
 }) => {
-  const [isMenuVisible, setMenuVisibility] = useState(false);
+  const { onPressDropdownButton, isDropdownOpen, closeOpenedDropdown } =
+    useDropdowns();
+  const dropdownRef = useRef<View>(null);
+
   return (
     <View style={{ position: "relative" }}>
-      <TouchableOpacity onPress={() => setMenuVisibility((prev) => !prev)}>
+      <TouchableOpacity onPress={() => onPressDropdownButton(dropdownRef)}>
         {component}
       </TouchableOpacity>
-      {isMenuVisible && (
-        <PrimaryBox
-          width={width}
-          style={{ position: "absolute", right: 0, bottom: -20 }}
-          mainContainerStyle={{
-            position: "absolute",
-            paddingHorizontal: 12,
-          }}
-        >
-          {items.map((item, index) => (
-            <TouchableOpacity
-              disabled={item.disabled}
-              key={item.label}
-              onPress={() => {
-                setMenuVisibility(false);
-                item.onPress();
-              }}
-              activeOpacity={0.7}
-              style={[
-                { paddingVertical: 12, width: "100%" },
-                index !== items.length - 1 && {
-                  borderBottomWidth: 1,
-                  borderColor: neutral33,
-                },
-              ]}
-            >
-              <BrandText
+      {isDropdownOpen(dropdownRef) && (
+        <View ref={dropdownRef}>
+          <PrimaryBox
+            width={width}
+            style={{ position: "absolute", right: 0, bottom: -20 }}
+            mainContainerStyle={{
+              position: "absolute",
+              paddingHorizontal: 12,
```

Comment:
```suggestion
              paddingHorizontal: layout.spacing_x1_5,
```
We use variables for paddings and margins to keep consistency (Don't forget to import)